digitaltwin_client/inc/digitaltwin_device_client.h

```diff
@@ -75,7 +76,7 @@ MOCKABLE_FUNCTION(, DIGITALTWIN_CLIENT_RESULT, DigitalTwin_DeviceClient_CreateFr
  
   @returns  A DIGITALTWIN_CLIENT_RESULT.
 */
-MOCKABLE_FUNCTION(, DIGITALTWIN_CLIENT_RESULT, DigitalTwin_DeviceClient_RegisterInterfacesAsync, DIGITALTWIN_DEVICE_CLIENT_HANDLE, dtDeviceClientHandle, const char*, deviceCapabilityModel, DIGITALTWIN_INTERFACE_CLIENT_HANDLE*, dtInterfaces, unsigned int, numDTInterfaces, DIGITALTWIN_INTERFACE_REGISTERED_CALLBACK, dtInterfaceRegisteredCallback, void*, userContextCallback);
+MOCKABLE_FUNCTION(, DIGITALTWIN_CLIENT_RESULT, DigitalTwin_DeviceClient_RegisterInterfacesAsync, DIGITALTWIN_DEVICE_CLIENT_HANDLE, dtDeviceClientHandle, DIGITALTWIN_INTERFACE_CLIENT_HANDLE*, dtInterfaces, unsigned int, numDTInterfaces, DIGITALTWIN_INTERFACE_REGISTERED_CALLBACK, dtInterfaceRegisteredCallback, void*, userContextCallback);
```

Comment:
>DIGITALTWIN_INTERFACE_REGISTERED_CALLBACK [](start = 236, length = 41)

Just occurred to me but I don't think we need DIGITALTWIN_INTERFACE_REGISTERED_CALLBACK anymore, do we?  There's nothing truly async as far as caller is concerned.  So we could remove the dt_client_core.c's DT_REGISTER_INTERFACES_CALLBACK_CONTEXT stuff also.  We still would want to leave the callbacks into interfaces DT_InterfaceList_RegistrationCompleteCallback() around though.  If we remove this callback, it also simplifies the samples so there's not the polling.

Reply:
I have preliminarily removed DIGITALTWIN_INTERFACE_REGISTERED_CALLBACK and related context stuff for your review.  A caveat is that the application do not get any insight/notification of the state of connection, thus interfaces would start operation despite mqtt connect failure.  If we determined that this is by-design for refresh, I will make additional change in sample LL and e2e to remove APP_DIGITALTWIN_REGISTRATION_STATUS_TAG related code similar to what I have already done in digitialtwin_sample_device.c.

Reply:
Per offline email, I think this is right.  Next time I'm in we can discuss more f2f.

---
In reply to: [386687567](https://github.com/Azure/azure-iot-sdk-c/pull/1439#discussion_r386687567) [](ancestors = 386687567)